src/org/skypro/skyshop/product/Product.java

```diff
@@ -36,9 +36,14 @@ public String getSearchableName() {
     }
     @Override
     public boolean equals(Object obj) {
-        if (this == obj) return true;
-        if (obj == null || obj instanceof Searchable) {
-            Product other = (Product) obj;
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (!(obj instanceof Product)) {
```

Comment:
```suggestion
        if (obj instanceof Product) {
```
наоборот, если тип Product, то можно к нему привести